homeassistant/components/aemet/weather_update_coordinator.py

```diff
@@ -0,0 +1,652 @@
+        return None
+
+    @staticmethod
+    def _get_temperature_day(day_data):
+        """Get temperature (day) from weather data."""
+        val = get_forecast_day_value(
+            day_data[AEMET_ATTR_TEMPERATURE], key=AEMET_ATTR_MAX
+        )
+        if val:
+            return format_int(val)
+        return None
+
+    @staticmethod
+    def _get_temperature_low_day(day_data):
+        """Get temperature (day) from weather data."""
+        val = get_forecast_day_value(
+            day_data[AEMET_ATTR_TEMPERATURE], key=AEMET_ATTR_MIN
+        )
+        if val:
+            return format_int(val)
+        return None
+
+    @staticmethod
+    def _get_temperature_feeling(day_data, hour):
+        """Get temperature from weather data."""
+        val = get_forecast_hour_value(day_data[AEMET_ATTR_TEMPERATURE_FEELING], hour)
+        if val:
+            return format_int(val)
+        return None
+
+    def _get_town_id(self):
+        """Get town ID from weather data."""
+        if self._town:
+            return self._town[AEMET_ATTR_ID]
+        return None
+
+    def _get_town_name(self):
+        """Get town name from weather data."""
+        if self._town:
+            return self._town[AEMET_ATTR_NAME]
+        return None
+
+    @staticmethod
+    def _get_wind_bearing(day_data, hour):
+        """Get wind bearing (hour) from weather data."""
+        val = get_forecast_hour_value(
+            day_data[AEMET_ATTR_WIND_GUST], hour, key=AEMET_ATTR_DIRECTION
+        )[0]
+        if val in WIND_BEARING_MAP:
+            return WIND_BEARING_MAP[val]
+        _LOGGER.error("%s not found in Wind Bearing map", val)
+        return None
+
+    @staticmethod
+    def _get_wind_bearing_day(day_data):
+        """Get wind bearing (day) from weather data."""
+        val = get_forecast_day_value(
+            day_data[AEMET_ATTR_WIND], key=AEMET_ATTR_DIRECTION
+        )
+        if val in WIND_BEARING_MAP:
+            return WIND_BEARING_MAP[val]
+        _LOGGER.error("%s not found in Wind Bearing map", val)
+        return None
+
+    @staticmethod
+    def _get_wind_max_speed(day_data, hour):
+        """Get wind max speed from weather data."""
+        val = get_forecast_hour_value(day_data[AEMET_ATTR_WIND_GUST], hour)
+        if val:
+            return format_int(val)
+        return None
+
+    @staticmethod
+    def _get_wind_speed(day_data, hour):
+        """Get wind speed (hour) from weather data."""
+        val = get_forecast_hour_value(
+            day_data[AEMET_ATTR_WIND_GUST], hour, key=AEMET_ATTR_SPEED
+        )[0]
+        if val:
+            return format_int(val)
+        return None
+
+    @staticmethod
+    def _get_wind_speed_day(day_data):
+        """Get wind speed (day) from weather data."""
+        val = get_forecast_day_value(day_data[AEMET_ATTR_WIND], key=AEMET_ATTR_SPEED)
+        if val:
+            return format_int(val)
+        return None
+
+
+class AemetWeather:
```

Comment:
This could be a dataclasses dataclass.

https://docs.python.org/3/library/dataclasses.html

Reply:
As above, this is inherited from OpenWeatherMap:
https://github.com/home-assistant/core/blob/fd177441b3cfddb22b893b26e91862307bbcc437/homeassistant/components/openweathermap/weather_update_coordinator.py#L215